Inital WaylandTexBuffer and mg::Texture interface

mariogrip opened the pull request:
This adds support for wayland support. 

For egl you need to set export EGL_PLATFORM=wayland and LD to use libhybris-egl libs

mariogrip committed:
Inital WaylandAllocator

mariogrip commented:
I also have this that hooks it up, https://github.com/ubports/mir-android-platform/commit/56eabacdc37784849fcdebc402c3dbeec374572d but not sure if this is the correct place to hook it up.

APokorny commented:
This essentially creates a platform for running a mir server as a client of a wayland compositor?
that then maybe also requires a wayland input platform, so that the server may receive user input...

mariogrip commented:
This just sets up the wayland buffers on the android side, mir handles input etc

APokorny commented on src/platforms/android/server/graphic_buffer_allocator.cpp:
```diff
@@ -138,3 +146,254 @@ std::shared_ptr<mg::Buffer> mga::GraphicBufferAllocator::alloc_buffer(
+
+    std::shared_ptr<mir::graphics::NativeBuffer> native_buffer_handle() const override
+    {
+        return nullptr;
```
this one needs to impemented - right?

mariogrip replied:
Its not implemented upstream so i would guess not yet. mesa platform returns nullptr too

mariogrip replied:
Humm, i guess you are right, we might need this

mariogrip committed:
Update wayland allocator to handle upstream changes, add on_release arg

mariogrip committed:
Hook up wayland allocator

mariogrip committed:
Make create_gl_context const to match upstream override

mariogrip committed:
Pass a `Display const&` through to `create_buffer_allocator` to match override

mariogrip committed:
Ignore downcast throw to android::NativeBuffer

This may be wayland, we simply have to ignore them since they cannot be
downcasted, and should not call any android funcs.

mariogrip committed:
Provide new mg::Texture interface

This provides the new mg::Texture and removed the old WaylandBuffer and
replaces it with WaylandTexBuffer

mariogrip committed:
Merge branch 'master' into wayland-fun

mariogrip commented:
Now both shm and egl work

egl still seems to have problems where it sometimes reports 0,0 buffer size, but this seems to be a problem upstream: https://github.com/MirServer/mir/blob/master/src/server/frontend_wayland/wl_surface.cpp#L353

For egl you need to set `export EGL_PLATFORM=wayland` and LD to use libhybris-egl libs

UniversalSuperBox commented:
Wow, it's been a while. Does anything need to happen with this any more?
